src/events/google_cloud.js

```diff
@@ -79,6 +87,132 @@ function bigQueryWrapper(wrappedFunction) {
+                    if ((!pubsubProjectId || pubsubProjectId === GOOGLE_CLOUD_TYPES
+                        .defaultProjectId) && !!requestFunctionThis.projectId) {
+                        event.getResource().setName(requestFunctionThis.projectId);
+                    }
+                    const callbackResponse = {};
+                    if (arg2) {
+                        if (config.method === 'publish') {
+                            const messageIds = arg2.messageIds ? arg2.messageIds : arg2;
+                            if (Array.isArray(messageIds) && messageIds.length) {
+                                callbackResponse.messageIds = messageIds;
+                                const [messageId] = messageIds;
+                                callbackResponse.messageId = messageId;
+                            }
+                        } else if (config.method === 'createSubscription') {
+                            callbackResponse.subscription = arg2;
+                        } else if (config.method === 'deleteSubscription') {
+                            if (config.reqOpts && config.reqOpts.subscription) {
+                                callbackResponse.subscription = getLastSplittedString(
+                                    config.reqOpts.subscription
+                                );
+                            }
+                        } else if (config.method === 'deleteTopic') {
+                            if (config.reqOpts && config.reqOpts.topic) {
+                                callbackResponse.topic = getLastSplittedString(
+                                    config.reqOpts.topic
+                                );
+                            }
+                        } else if (config.method === 'createTopic') {
+                            callbackResponse.topic = arg2;
+                        }
+                    }
+                    finalizeEvent(event, startTime, err, callbackResponse);
+                    resolve();
+                    if (callback) {
+                        if (arg3) {
+                            callback(err, arg2, ...arg3);
+                        } else {
+                            callback(err, arg2);
+                        }
+                    }
+                };
+            });
+            tracer.addEvent(event, responsePromise);
+        } catch (err) {
+            tracer.addException(err);
+        }
+        return original.apply(this, [config, patchedCallback]);
+    };
+}
+
+/**
+ * Wrap pubsub request function.
+ * @param {Function} original pubsub request function.
+ * @returns {Function} The wrapped function
+ */
+function wrapPubSubPullFunction(original) {
+    return function internalPubSubPullFunction(request, options, callback) {
+        try {
+            let pubsubProjectId;
+            if (request && request.subscription) {
+                const subscriptionSplited = request.subscription.split('/');
```

Comment:
you have a utility function for this

Reply:
yes, to get the last item. here I want to get the second item.